### PR TITLE
IBU-10830

### PR DESCRIPTION
**Pull request by HobnobMancer**
[Jira](https://www.ebi.ac.uk/panda/jira/browse/IBU-10830)

## Update release workflow: `interpro.xml`

* Add `is-llm` and `is-llm-reviewed` attributes to XML output.
* Add `is-llm` and `is-llm-reviewed` child nodes to abstract attribute.
    * `is-llm` set to true if at least one description is ai-generated
    * `is-llm-reviewed` set to true if all ai-generated abstracts are reviewed

### Marked as draft

Need to fix existing `AttributeError` in `xmlfiles.py`:

```bash
Traceback (most recent call last):
  File "...interpro7-dw/interpro7dw/interpro/ftp/run.py", line 4, in <module>
    export_interpro(
  File "...interpro7-dw/interpro7dw/interpro/ftp/xmlfiles.py", line 447, in export_interpro
    elem.writexml(fh, addindent="  ", newl="\n")
  File "...python3.11/xml/dom/minidom.py", line 897, in writexml
    node.writexml(writer, indent+addindent, addindent, newl)
  File "...python3.11/xml/dom/minidom.py", line 897, in writexml
    node.writexml(writer, indent+addindent, addindent, newl)
  File "...python3.11/xml/dom/minidom.py", line 886, in writexml
    _write_data(writer, attrs[a_name].value)
  File "...python3.11/xml/dom/minidom.py", line 306, in _write_data
    data = data.replace("&", "&amp;").replace("<", "&lt;"). \
           ^^^^^^^^^^^^
AttributeError: 'tuple' object has no attribute 'replace'
```

### CODE REVIEW

File: interpro7dw/interpro/ftp/xmlfiles.py [hunk 1]
```diff
@@ -234,23 +239,43 @@ def export_interpro(entries_file: str, entry2xrefs_file: str,
...
 
+                _abstract_llm = doc.createElement("abstract-llm")
+                _abstract_llm.appendChild(doc.createTextNode(abstract_is_llm))
+                abstract.appendChild(_abstract_llm)
+
+                _abstract_llm_reviewed = doc.createElement("abstract-llm-reviewed")
+                _abstract_llm_reviewed.appendChild(doc.createTextNode(abstract_is_llm_reviewed))
+                abstract.appendChild(_abstract_llm_reviewed)
+        
+
```
Comment (matthiasblum) [inline interpro7dw/interpro/ftp/xmlfiles.py:278]: Just add attributes to the `abstract` element:

```suggestion
                abstract.setAttribute("is-llm", ab_is_llm)
                abstract.setAttribute("is-llm-reviewed", ab_is_rev_llm)
                elem.appendChild(abstract)

```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
I added a few suggestions.
You'll also need to change the target branch to `dev` (there will be conflicts, feel free to have a look but if you're unsure, let me know and I'll resolve them).
Thanks :)

File: interpro7dw/interpro/ftp/xmlfiles.py [hunk 2]
```diff
@@ -236,21 +241,43 @@ def export_interpro(entries_file: str, entry2xrefs_file: str,
...
+                        reviewed_llm_descrs += 1
+
+            text = _restore_abstract("\n".join(blocks))
+            ab_is_llm = ab_is_reviewed_llm = "false"
+            if llm_descrs > 0:
+                # At least one AI-generated description
+                ab_is_llm = "true"
+                if reviewed_llm_descrs == llm_descrs:
+                    # Considered reviewed if all AI descriptions are reviewed
+                    ab_is_reviewed_llm = "true"
```
Comment (matthiasblum) [inline interpro7dw/interpro/ftp/xmlfiles.py:268]: Fix `NameError`:

```suggestion
                abstract.setAttribute("is-llm-reviewed", ab_is_reviewed_llm)
```

File: interpro7dw/interpro/ftp/xmlfiles.py [hunk 3]
```diff
@@ -236,21 +241,43 @@ def export_interpro(entries_file: str, entry2xrefs_file: str,
...
             except ExpatError as exc:
                 # TODO: use CDATA section for all entries
-                logger.warning(f"{entry_acc}: {exc}")
+                logger.warning(f"{entry_acc}: {exc} -- -*-")
                 # abstract = doc.createElement("abstract")
                 # abstract.appendChild(doc.createCDATASection(text))
             else:
                 abstract = _doc.documentElement
+                abstract.setAttribute("is-llm", ab_is_llm)
+                abstract.setAttribute("is-llm-reviewed", ab_is_rev_llm)
```
Comment (matthiasblum) [inline interpro7dw/interpro/ftp/xmlfiles.py:280]: Fix `NameError`:

```suggestion
                abstract.setAttribute("is-llm-reviewed", ab_is_reviewed_llm)
```

**Comment by HobnobMancer** [comment]
Tested, and XML file generated successfully and correctly.

**Pull request merged**